File: grind75/week2/026-Daily-Temperatures-Medium/026-Daily-Temperatures-Medium.py
# link: https://leetcode.com/problems/daily-temperatures/
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        my_stack = []
        ans = [0] * len(temperatures)

        for i in range(len(temperatures)):

            while my_stack and temperatures[i] > temperatures[my_stack[-1]]:
                idx = my_stack.pop()
                ans[idx] = i - idx
            my_stack.append(i)
            
        return ans
        

File: grind75/week2/026-Daily-Temperatures-Medium/026-Daily-Temperatures-Medium.cpp
// link: https://leetcode.com/problems/daily-temperatures/
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> my_stack;
        vector<int> ans(temperatures.size(), 0);

        for (int i = 0; i < temperatures.size(); i++) 
        {
            while (!my_stack.empty() && temperatures[i] > temperatures[my_stack.top()]) 
            {
                int idx = my_stack.top();
                my_stack.pop();
                ans[idx] = i - idx;
            }
            my_stack.push(i);
        }

        return ans;
    }
};
